aws/components/apigateway/setup.ftl

```diff
@@ -224,17 +238,36 @@
             )
         ] ]
     [#if apiPolicyCidr?has_content ]
+        [#-- If lambda authorizers are in use, and the default AuthorisationModel is --]
+        [#-- effect, warn and force the model to a valid value                       --]
+        [#if lambdaAuthorizers?has_content && !apiPolicyAuth?starts_with("AUTHORI") ]
+            [@warn
+                message="Authorization model of \"" + apiPolicyAuth + "\" is not compatible with the use of lambda authorizers. Forcing to \"AUTHORISER_AND_IP\"."
+            /]
+            [#local apiPolicyAuth = "AUTHORISER_AND_IP" ]
+        [/#if]
+
+        [#-- If cognito authorizers are in use, the AuthorisationModel must be IP --]
+        [#if cognitoPools?has_content && (apiPolicyAuth != "IP") ]
+            [@warn
+                message="Authorization model of \"" + apiPolicyAuth + "\" is not compatible with the use of cognito authorizers. Forcing to \"IP\"."
+            /]
+            [#local apiPolicyAuth = "IP" ]
+        [/#if]
```

Comment:
Should we make these fatal and have people change them. If the configuration option is saying it should be this value then we should probably have people change it in the config.

Reply:
I guess that wouldn't be a breaking change in that it will fix a bug, but it will break previously "working" configurations. However it will only kick in with the update that fixes the bug. 

On balance I think you are right - I'll tweak them.